traffic/speed_groups.cpp

```diff
@@ -1,9 +1,24 @@
 #include "traffic/speed_groups.hpp"
 
+#include "std/algorithm.hpp"
+
 namespace traffic
 {
 uint32_t const kSpeedGroupThresholdPercentage[] = {8, 16, 33, 58, 83, 100, 100, 100};
 
+SpeedGroup GetSpeedGroupByPercentage(double p)
+{
+  p = min(p, 100.0);
+  uint32_t const pu = static_cast<uint32_t>(p);
```

Comment:
Why not? One explicit cast instead of several implicit.
Besides, it makes clear the intention to round the value before comparison.